exploring.rst

```diff
@@ -0,0 +1,68 @@
+.. _exploring:
+
+Exploring the CPython's Internals
+=================================
+
+This is a quick guide for who want to know about CPython's internals,
+and provide a summary about the source code structure and references.
+
+
+CPython Source Code Typical Layout
+---------------------------------------
+
+For Python modules, the typical layout is:
+
+* ``Lib/<module>.py``
+* ``Modules/_<module>module.c`` (if there's also a C accelerator module)
+* ``Lib/test/test_<module>.py``
+* ``Doc/library/<module>.rst``
+
+For extension-only modules, the typical layout is:
+
+* ``Modules/<module>module.c``
+* ``Lib/test/test_<module>.py``
+* ``Doc/library/<module>.rst``
+
+For builtin types, the typical layout is:
+
+* ``Objects/<builtin>object.c``
+* ``Lib/test/test_<builtin>.py``
+* ``Doc/library/stdtypes.rst``
+
+For builtin functions, the typical layout is:
+
+* ``Python/bltinmodule.c``
+* ``Lib/test/test_<builtin>.py``
+* ``Doc/library/functions.rst``
+
+Some Exception:
+
+* builtin type ``int`` is at ``Objects/longobject.c``
+* builtin type ``str`` is at ``Objects/unicodeobject.c``
+
+CPython Internals Tutorials
+---------------------------
+
+There are several materials about CPython internals on the Internet:
+
+* Python 2
```

Comment:
@willingc, I would like to change to use table and point out each material based version, to remind reader that the article will somehow out of date.